pkg/iac-providers/kustomize/commons/load-dir.go

```diff
@@ -0,0 +1,163 @@
+		return allResourcesConfig, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: errMsg})
+	}
+
+	kustomizeFileName := *files[0]
+	yamlkustomizeobj, err := utils.ReadYamlFile(filepath.Join(t.absRootDir, kustomizeFileName))
+
+	if err != nil {
+		err = fmt.Errorf("unable to read the kustomization file in the directory %s, error: %v", t.absRootDir, err)
+		zap.S().Error("error while reading the file", kustomizeFileName, zap.Error(err))
+		return allResourcesConfig, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: err.Error()})
+	}
+
+	// ResourceConfig representing the kustomization.y(a)ml file
+	config := output.ResourceConfig{
+		Name:   filepath.Dir(t.absRootDir),
+		Type:   kustomizedirectory,
+		Line:   1,
+		ID:     kustomizedirectory + "." + filepath.Dir(t.absRootDir),
+		Source: filepath.Join(t.absRootDir, kustomizeFileName),
+		Config: yamlkustomizeobj,
+	}
+
+	allResourcesConfig[kustomizedirectory] = append(allResourcesConfig[kustomizedirectory], config)
+
+	// obtaining list of IacDocuments from the target working directory
+	iacDocuments, err := LoadKustomize(t.absRootDir, kustomizeFileName, t.version, t.useKustomizeBinary)
+	if err != nil {
+		errMsg := fmt.Sprintf("error occurred while loading kustomize directory '%s'. err: %v", t.absRootDir, err)
+		zap.S().Error("error occurred while loading kustomize directory", zap.String("kustomize directory", t.absRootDir), zap.Error(err))
+		return nil, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: errMsg})
+	}
+
+	for _, doc := range iacDocuments {
+		var k k8sv1.K8sV1
+		var config *output.ResourceConfig
+
+		config, err = k.Normalize(doc)
+		if err != nil {
+			zap.S().Warn("unable to normalize data", zap.Error(err), zap.String("file", doc.FilePath))
+			continue
+		}
+
+		// TODO finding a better solution to detect accurate line number for tracing back the files causing violations
+		config.Line = 1
+		config.Source = doc.FilePath
+		allResourcesConfig[config.Type] = append(allResourcesConfig[config.Type], *config)
+	}
+
+	return allResourcesConfig, t.errIacLoadDirs
+}
+
+// LoadKustomize loads up a 'kustomized' directory and returns a returns a list of IacDocuments
+func LoadKustomize(basepath, filename, version string, useKustomizeBinary bool) ([]*utils.IacDocument, error) {
+
+	var yaml []byte
+	if useKustomizeBinary {
+		envVar := "KUSTOMIZE_" + version
+		kustomizeBinaryPath, ok := os.LookupEnv(envVar)
+		if !ok {
+			return nil, fmt.Errorf(kustomizeErrMessage, "Environment variable "+envVar+" not set")
+		}
+		exe := kustomizeBinaryPath
+		arg1 := "build"
+		arg2 := basepath
+
+		cmd := exec.Command(exe, arg1, arg2)
+
+		var stdOut, stdErr bytes.Buffer
+		cmd.Stdout = &stdOut
+		cmd.Stderr = &stdErr
+
+		err := cmd.Run()
+
+		if stdErr.String() != "" {
```

Comment:
Checking for data in stdErr and exiting could cause failures when there are none... Log data is often written to stderr, for example. Recommend removing this and just relying on the error code (next if statement)

Reply:
Thanks for pointing this out. I am using stdErr to get the `detailed error cause`, as `error` was returning only exit code. But yeah we should not return just because we get the stdErr.